spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc

```diff
@@ -1159,4 +1160,72 @@ Spring Cloud Data Flow will have a suffix appended to it consisting of `-scdf-<t
+
+Versions of an application are managed by registering multiple applications with the same name and coordinates _except_ the version.  For example, if you were to register an application with the following values, you would get one application registered with two versions (2.0.0.RELEASE and 2.1.0.RELEASE):
+
+Application 1
+
+* Name: `timestamp`
+* Type: `task`
+* URI: `maven://org.springframework.cloud.task.app:timestamp-task:2.0.0.RELEASE`
+
+Application 2
+
+* Name: `timestamp`
+* Type: `task`
+* URI: `maven://org.springframework.cloud.task.app:timestamp-task:2.1.0.RELEASE`
+
+Besides having multiple versions, Spring Cloud Data Flow needs to know which version to execute on the next launch.  This is indicated by setting a version to be the default version.  Whatever version of a task application is configured as the default version will be the one to be executed on the next launch request.  You can see which version is the default in the UI as shown below.
+
+image::{dataflow-asciidoc}/images/dataflow-task-default-version.png[Task Application Default Version, scaledwidth="50%"]
+
+=== Task Launch Lifecycle
+In previous versions of Spring Cloud Data Flow, when the request to launch a task was received, Spring Cloud Data Flow would deploy the application (if needed) and execute it.  If the application was being run on a plaform that did not need to have the application deployed every time (CloudFoundry for example), the previously deployed application was used.  This flow has changed in 2.3.  Let's take a look at what happens when a task launch request comes in now.
+
+image::{dataflow-asciidoc}/images/dataflow-task-launch-flow.png[Flow For Launching A Task, scaledwidth="50%"]
+
+There are three main flows to consider in the above diagram.  Launching the first time/launching with no changes is one, the other is launching when there are changes.  We'll look at the flow with no changes first.
+
+==== Launch a Task With No Changes
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is not required since nothing has changed (no properties, deployment properites, or versions have changed since the last execution).
+
+[start=5]
+5. On platforms that cache a deployed artifact (CloudFoundry at the writing of this documentation), Data Flow checks if the application was previously deployed.
+6. If the application needs to be deployed, Data Flow will do the deployment of the task application.
+7. Data Flow launches the application.
+
+That flow is the default behavior and will occur every time a request comes in if nothing has changed.  It is important to note that this is the same flow that Data Flow has always executed for launching of tasks.
+
+==== Launch a Task With Changes That Is Not Currently Running
+The second flow to consider when launching a task is if there was a change in any of the task application version, application properties, or deployment properties.  In this case, the following flow is executed
```

Comment:
1. Is that a leading space before the paragraph? If so, delete it. Replace "if" with "whether".

"If" and "whether" are sometimes interchangeable, but it's not generally a good idea to use "if" for anything other than conditional clauses. If something may or may not have happened or something may or may not be true, "whether" is the way to go. In this particular sentence, I had to read the sentence twice, because "if" made me misread it the first time. So "whether" offers greater clarity.

2. Add a colon at the end of the sentence (to lead into the procedure).